core/src/forwarding_stage.rs

```diff
@@ -151,43 +161,63 @@ pub(crate) fn spawn_forwarding_stage(
         }
         ForwardingClientOption::TpuClientNext((
             stake_identity,
-            tpu_client_socket,
+            tpu_client_sockets,
             runtime_handle,
             cancel,
+            node_multihoming,
         )) => {
-            let non_vote_client = TpuClientNextClient::new(
-                runtime_handle,
-                forward_address_getter,
-                Some(stake_identity),
-                tpu_client_socket,
-                cancel,
-            );
+            // Create TPU clients for each socket provided.
+            // Number of clients is same as number of bind IP addresses.
+            let non_vote_clients: Box<[TpuClientNextClient]> = tpu_client_sockets
+                .into_vec()
```

Comment:
I wonder if we need `Box` if we later convert it to `Vec` anyways.

Reply:
I want this to be immutable with good reason to ensure number of entries lines up with number of interfaces. Box<[]> achieves that.